solver/src/interactions/uniswap.rs

```diff
@@ -0,0 +1,83 @@
+use crate::{
+    encoding::{self, INTERACTION_BASE_SIZE},
+    settlement::Interaction,
+};
+use anyhow::Result;
+use contracts::UniswapV2Router02;
+use primitive_types::{H160, U256};
+
+const UNISWAP_DATA_SIZE: usize = 260;
+const UNISWAP_TOTAL_SIZE: usize = INTERACTION_BASE_SIZE + UNISWAP_DATA_SIZE;
+
+#[derive(Debug)]
+pub struct UniswapInteraction {
+    pub uniswap_router_address: H160,
+    pub amount_in: U256,
+    pub amount_out_min: U256,
+    pub token_in: H160,
+    pub token_out: H160,
+    pub payout_to: H160,
+}
+
+impl Interaction for UniswapInteraction {
+    fn encode(&self, writer: &mut dyn std::io::Write) -> Result<()> {
+        let uniswap = UniswapV2Router02::at(&super::dummy_web3::dummy_web3(), H160::zero());
+        let method = uniswap.swap_exact_tokens_for_tokens(
+            self.amount_in,
+            self.amount_out_min,
+            vec![self.token_in, self.token_out],
+            self.payout_to,
+            U256::MAX,
+        );
+        let data = method.tx.data.expect("no calldata").0;
+        writer.write_all(self.uniswap_router_address.as_fixed_bytes())?;
+        // Unwrap because we know uniswap data size can be stored in 3 bytes.
+        writer.write_all(&encoding::encode_interaction_data_length(UNISWAP_DATA_SIZE).unwrap())?;
```

Comment:
Should we use `data.len()` instead of `UNISWAP_DATA_SIZE`? We could also assert they are equal but I don't really think there is value in knowing the exact length of every interaction (other than for the purpose of testing).

Reply:
switched to data.len()